File: packages/room-server/src/database/datasheet/entities/datasheet.changeset.entity.ts
import { IOperation } from '@apitable/core';
import { BaseEntity } from 'shared/entities/base.entity';
import { Column, Entity } from 'typeorm';

/**
 * Workbench-Digital table operation change collection table
 */
@Entity('datasheet_changeset')
export class DatasheetChangesetEntity extends BaseEntity {
  @Column({
    name: 'message_id',
    nullable: true,
    comment: 'Unique identifier of a changeset request',
    length: 255,
  })
  messageId?: string;

  @Column({
    name: 'dst_id',
    nullable: true,
    comment: 'datasheet ID',
    length: 50,
  })
  dstId?: string;

  @Column({
    name: 'member_id',
    nullable: true,
    comment: 'Operating member ID (associated#organization_member#ID)',
  })
  memberId?: string;

  @Column('json', {
    name: 'operations',
    nullable: true,
    comment: 'Operation Action collection',
  })
  operations?: IOperation[];

  @Column({
    name: 'revision',
    nullable: true,
    comment: 'revision',
    unsigned: true,
    default: () => 0,
  })
  revision!: number;
}

File: backend-server/application/src/main/java/com/apitable/workspace/entity/DatasheetChangesetEntity.java


package com.apitable.workspace.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * Workbench - Data Table Operation Changeset Collection Table.
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "datasheet_changeset")
public class DatasheetChangesetEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary key.
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * The unique ID of the changeset request, which is used to ensure the uniqueness of the changeset.
     */
    private String messageId;

    /**
     * Datasheet ID.
     */
    private String dstId;

    /**
     * Action member ID(link#xxxx_organization_member#id).
     */
    private Long memberId;

    /**
     * Collection of operation actions.
     */
    private String operations;

    /**
     * Version No.
     */
    private Long revision;

    /**
     * 1:Delete，0:Not Deleted.
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * Create User.
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * Last Update User.
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * Creation time.
     */
    private LocalDateTime createTime;

    /**
     * Update time.
     */
    private LocalDateTime updateTime;

    /**
     * Create Time.
     */
    private LocalDateTime createdAt;

    /**
     * Update Time.
     */
    private LocalDateTime updatedAt;

}
